tests/devices/qutrit_mixed/test_qutrit_mixed_simulate.py

```diff
@@ -0,0 +1,735 @@
+
+        assert list(debugger.snapshots.keys()) == [0, "final_state"]
+        assert qml.math.allclose(debugger.snapshots[0], self.basis_state)
+
+        expected_final_state = math.asarray(
+            expected_TRX_circ_state(phi.detach().numpy(), subspace), like="torch"
+        )
+        assert qml.math.allclose(debugger.snapshots["final_state"], expected_final_state)
+
+    # pylint: disable=invalid-unary-operand-type
+    @pytest.mark.tf
+    def test_debugger_tf(self, subspace):
+        """Tests debugger with tensorflow."""
+        import tensorflow as tf
+
+        phi = tf.Variable(4.873)
+        debugger = self.Debugger()
+
+        qs = self.get_debugger_quantum_script(phi, subspace)
+        result = simulate(qs, debugger=debugger)
+
+        expected = expected_TRX_circ_expval_values(phi, subspace)
+        assert qml.math.allclose(result, expected)
+
+        assert list(debugger.snapshots.keys()) == [0, "final_state"]
+        assert qml.math.allclose(debugger.snapshots[0], self.basis_state)
+
+        expected_final_state = expected_TRX_circ_state(phi, subspace)
+        assert qml.math.allclose(debugger.snapshots["final_state"], expected_final_state)
+
+
+@pytest.mark.parametrize("subspace", [(0, 1), (0, 2)])
+class TestSampleMeasurements:
+    """Tests circuits with sample-based measurements"""
+
+    @staticmethod
+    def expval_of_TRY_circ(x, subspace):
+        """Find the expval of GellMann_3 on simple TRY circuit"""
+        if subspace[1] == 1:
+            return np.cos(x)
+        return np.cos(x / 2) ** 2
+
+    @staticmethod
+    def sample_sum_of_TRY_circ(x, subspace):
+        """Find the expval of computational basis for both wires on simple TRY circuit"""
+        if subspace[1] == 1:
+            return [np.sin(x / 2) ** 2, 0]
+        return [2 * np.sin(x / 2) ** 2, 0]
+
+    @staticmethod
+    def expval_of_2_qutrit_circ(x, subspace):
+        """expval of GellMann_3 on wire=0 on the 2 qutrit circuit used"""
+        if subspace[1] == 1:
+            return np.cos(x)
+        return np.cos(x / 2) ** 2
+
+    @staticmethod
+    def probs_of_2_qutrit_circ(x, y, subspace):
+        """possible measurement values and probabilityies for the 2 qutrit circuit used"""
+        probs = np.array(
+            [
+                np.cos(x / 2) * np.cos(y / 2),
+                np.cos(x / 2) * np.sin(y / 2),
+                np.sin(x / 2) * np.sin(y / 2),
+                np.sin(x / 2) * np.cos(y / 2),
+            ]
+        )
+        probs **= 2
+        if subspace[1] == 1:
+            keys = ["00", "01", "10", "11"]
+        else:
+            keys = ["00", "02", "20", "22"]
+        return keys, probs
+
+    def test_single_expval(self, subspace):
+        """Test a simple circuit with a single expval measurement"""
+        x = np.array(0.732)
+        qs = qml.tape.QuantumScript(
+            [qml.TRY(x, wires=0, subspace=subspace)],
+            [qml.expval(qml.GellMann(0, 3))],
+            shots=1000000,
```

Comment:
Is it necessary to use such a high number? `atol` is set to 0.1.

Reply:
I think that was a mistake.